package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua

```diff
@@ -38,15 +38,47 @@ f:section(Section, nil, translate(
 
 local mesh_vifs_5ghz = {}
 
+local function add_or_remove_role(roles, role, disabled)
+	if disabled == true then
+		util.remove_from_set(roles, role)
+	elseif disabled == false then
+		util.add_to_set(roles, role)
+	end
+end
+
+local function vif_option(section, role_name, band, msg)
+	local config_roles = uci:get_list('gluon', band, 'role')
+	local o = section:option(Flag, band .. '_' .. role_name .. '_enabled', msg)
+	o.default = util.contains(config_roles, role_name)
+
+	function o:write(data)
+		-- this does race for multiple options without additional read before write in o:write
+		local roles = uci:get_list('gluon', band, 'role')
+		add_or_remove_role(roles, role_name, not data)
+		uci:set('gluon', band, 'role', roles)
+	end
+
+	return o
+end
+
+local bands = {}
 
 uci:foreach('wireless', 'wifi-device', function(config)
```

Comment:
Per-band and per-device config should be separated completely, iterating over bands instead of devices for the per-band config. Most config is per-band, with just things like txpower that aren't reflected in /etc/config/gluon yet remaining per-device.